src/probnum/filtsmooth/statespace/discrete_transition.py

```diff
@@ -350,34 +355,44 @@ def _backward_rv_sqrt(
             else:
                 gain = np.zeros((len(rv.mean), len(rv_obtained.mean)))
```

Comment:
this used to be the culprit for the bug (together with the below -- see my comment)